headers/xdp/prog_dispatcher.h

```diff
@@ -5,7 +5,15 @@
 
 #include <linux/types.h>
 
-#define XDP_METADATA_SECTION "xdp_metadata"
+#define XDP_METADATA_SECTION_ORIG "xdp_metadata"
+#define XDP_METADATA_SECTION_NEW ".data.xdp_metadata"
+
+#if defined(USE_LIBBPF_V1_SECTION_NAMES)
+#define XDP_METADATA_SECTION XDP_METADATA_SECTION_NEW
+#else
+#define XDP_METADATA_SECTION XDP_METADATA_SECTION_ORIG
+#endif
+
```

Comment:
Duplicating the dispatcher_version line results in:

```
xdp-dispatcher.c:201:52: error: section does not match previous declaration [-Werror,-Wsection]
__uint(dispatcher_version, XDP_DISPATCHER_VERSION) SEC(XDP_METADATA_SECTION_NEW)
```

So it would also require a different variable name to avoid the collision.

But I was thinking that the libxdp and dispatcher version on a system would match and would either be old or new, so only need the dispatcher_version declared in 1 section. It's legacy and new programs that we'd need to be able to read old or new metadata from. Or am I missing something?